src/main/java/online/book/store/controller/CartController.java

```diff
@@ -0,0 +1,55 @@
+package online.book.store.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import online.book.store.dto.cart.CartItemDeleteBookDto;
+import online.book.store.dto.cart.ShoppingCartAddBookDto;
+import online.book.store.dto.cart.ShoppingCartDto;
+import online.book.store.dto.cart.request.ItemCartAddBookRequestDto;
+import online.book.store.service.CartService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Cart management", description = "Endpoints for managing carts")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/cart")
+public class CartController {
+    private final CartService cartService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping
+    @Operation(summary = "Get a shopping cart", description = "Get a shopping cart by user id")
+    public ShoppingCartDto findByUserId(Long id) {
+        return cartService.findByUserId(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping
+    @Operation(summary = "Add a book", description = "Add a book to shopping cart")
+    public ShoppingCartAddBookDto addBook(
+            ShoppingCartAddBookDto shoppingCartDto,
+            ItemCartAddBookRequestDto itemCartDto) {
+        return cartService.addBook(shoppingCartDto, itemCartDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping
+    @Operation(summary = "Update a cart item", description = "Update a cart item by id")
+    public void updateCartItem(Long id, int quantity) {
```

Comment:
```suggestion
    public void updateCartItem(Authentication authentication,
                                              @PathVariable @Positive Long cartItemId,
                                              @RequestBody @Valid UpdateCartItemDto item) {
```

This method should return updated shopping cart